stm32-modules/heater-shaker/src/CMakeLists.txt

```diff
@@ -3,6 +3,30 @@
 # library. It is included in both host and cross configurations.
 
 find_package(Python)
+find_package(Git QUIET)
+
+# This is like version for project but works when there aren't any tags yet
+execute_process(COMMAND ${GIT_EXECUTABLE} describe --tags --match "heater-shaker*"
+  WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+  OUTPUT_VARIABLE TAGNAME
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+if (NOT TAGNAME)
+  set(TAGNAME "@(dev)-")
+endif()
+
+string(REGEX MATCH "@.*$" PREPENDED_VERSION ${TAGNAME})
+string(SUBSTRING ${PREPENDED_VERSION} 1 -l VERSION)
+if (NOT VERSION)
+  set(VERSION "(dev)")
+endif()
+execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
+  WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+  OUTPUT_VARIABLE HASH
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+set(heater-shaker_VERSION "${VERSION}-${HASH}")
```

Comment:
Is there a reason to make this a random mix of capital/lower case + kebab and snake case?

Reply:
to stay consistent with the project version utility, basically. and that's like that because appending `_VARIABLE` to some prefix is sort of the way (sadly) you do "return values" from cmake functions or macros